### ISSUE TITLE
No valid rules have been specified

### ISSUE THREAD

**Issue by nattyxd** [palantir/tslint#3735]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.7.2
- __Running TSLint via__: VScode on Mac, but fails on CLI too

#### With `tslint.json` configuration:

```json
{
    "defaultSeverity": "none",
    "extends": [
        "tslint:recommended"
    ],
    "jsRules": {},
    "rules": {
        "no-internal-module": false,
        "typedef-whitespace": false,
        "variable-name": [
            true,
            "check-format",
            "allow-leading-underscore"
        ],
        "no-console": false,
        "no-reference": false,
        "no-namespace": false,
        "max-line-length": [true, 140],
        "no-bitwise": false,
        "align": true,
        "arrow-return-shorthand": true,
        "no-magic-numbers": true
    },
    "rulesDirectory": []
}
```

#### Actual behavior
Last week my tslint was working fine with the above configuration. Today I updated my VSCode, and TSLint started complaining that no compatible version of Typescript was installed. To fix this I did `npm install -g typescript` which fixed the TSLint version error.

The TSLint extension in VSCode gives the warning `No valid rules have been specified`, and doesn't lint anything.

I tried running `tslint --project build` from the CLI, but I get the same error.

I thought maybe tslint was looking for a global configuration file, but if I add a syntax error to my local tslint.json, I'll get a different error `Unexpected token f`, so it seems it is looking at the local config but then failing somehow

#### Expected behavior
TSLint should run over the project as it did before

**Comment by liangchunn** [comment]
I have the same issue with coming from `5.8.0` to `5.9.1`.

Running `tslint --config tslint.json --project tsconfig.json` spits the same `No valid rules have been specified` error.

**Comment by liangchunn** [comment]
I've found a possible solution:
If you're using setting `"allowJs": true` in `tsconfig.json`, it will try to find `rules` **and** `jsRules` from `tslint.json`.

If your `jsRules` is empty, it will complain. IMO the messages should indicate which rules it is actually parsing.

Adding a rule inside `jsRules` made the message go away. 
```json
"jsRules": {
    "no-empty": true
}
```

**Comment by nattyxd** [comment]
I just tried adding:
```
"allowJs": true,
"jsRules": {
    "no-empty": true
}
```

To my tslint.json, but unfortunately the message stays, and it doesn't seem to lint anything still

**Comment by liangchunn** [comment]
What do you get on your output? If there’s no errors then it should be working fine.

**Comment by nattyxd** [comment]
I get the same output/warning, changing it didn't do anything 😢

**Comment by ajcrites** [comment]
@nattyxd did you add `jsRules` to your `tsconfig` or `tslint.json`? `allowJs` is in tsconfig; with `allowJs` your `tslint.json` is expecting some `jsRules` (hence the bug). A workaround would be to add `"jsRules": { "no-empty": true }` to your `tslint.json` so that it has a rule and doesn't complain. This shouldn't have any impact I think. It's just a small annoyance.

**Comment by liangchunn** [comment]
@ajcrites Absolutely on point. I think this issue should be closed.

**Comment by johnwiseheart** [comment]
Closing this issue as it seems to be resolved.

**Comment by gkirill** [comment]
Several months ago I had this issue, then it was solved by the solution explained in this thread, and it was all okey for a while, but recently I started having this issue again, though I do have `"no-empty": true`. Banging my head and cannot figure out what could be the problem.

**Comment by lonix1** [comment]
@gkirill Did you figure it out? I have the same problem. The advice above doesn't help.

**Comment by sdakoju** [comment]
{
  "extends": "@microsoft/sp-tslint-rules/base-tslint.json",
  "rules": {
    "class-name": false,
    "export-name": false,
    "forin": false,
    "label-position": false,
    "member-access": true,
    "no-arg": false,
    "no-console": false,
    "no-construct": false,
    "no-duplicate-variable": true,
    "no-eval": false,
    "no-function-expression": true,
    "no-internal-module": true,
    "no-shadowed-variable": true,
    "no-switch-case-fall-through": true,
    "no-unnecessary-semicolons": true,
    "no-unused-expression": true,
    "no-use-before-declare": true,
    "no-with-statement": true,
    "semicolon": true,
    "trailing-comma": false,
    "typedef": false,
    "typedef-whitespace": false,
    "use-named-parameter": true,
    "variable-name": false,
    "whitespace": false
  }
}

**Comment by liranh85** [comment]
If you don't want TSLint linting JS files (and you have tslint >= 5.8), simply add the following to your `tslint.json`:

```js
"linterOptions": {
    "exclude": [
        "**/*.js"
    ]
}
```

**Comment by TaylorPzreal** [comment]
Add ```tslint --config tslint.js``` works for me.

**Comment by danvln** [comment]
you need to specify the config file, like tslint -c .eslintrc.yml -p tsconfig.json, and there is nothing that's required from above like jsRules or all of those..

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
